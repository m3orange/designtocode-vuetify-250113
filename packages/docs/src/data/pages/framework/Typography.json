{
  "title": "header",
  "titleText": "headerText",
  "mdLink": "https://material.io/design/typography/the-type-system.html",
  "children": [
    {
      "type": "section",
      "children": [
        {
          "type": "heading",
          "lang": "helperHeader"
        },
        {
          "type": "text",
          "lang": "helperText"
        },
        {
          "type": "api",
          "value": [
            "v-display-4",
            "v-display-3",
            "v-display-2",
            "v-display-1",
            "v-headline",
            "v-title",
            "v-subtitle-2",
            "v-subtitle-1",
            "v-body-2",
            "v-body-1",
            "v-caption",
            "v-overline"
          ]
        }
      ]
    },
    {
      "type": "section",
      "children": [
        {
          "type": "heading",
          "lang": "fontHeader"
        },
        {
          "type": "text",
          "lang": "fontText"
        },
        {
          "type": "markdown",
          "lang": "fonts"
        },
        {
          "type": "style-headings"
        }
      ]
    },
    {
      "type": "section",
      "children": [
        {
          "type": "heading",
          "lang": "weightsHeader"
        },
        {
          "type": "text",
          "lang": "weightsText"
        },
        {
          "type": "markdown",
          "lang": "weights"
        },
        {
          "type": "text",
          "lang": "italicText"
        },
        {
          "type": "markdown",
          "lang": "italic"
        },
        {
          "type": "style-weights"
        }
      ]
    },
    {
      "type": "section",
      "children": [
        {
          "type": "heading",
          "lang": "transformHeader"
        },
        {
          "type": "text",
          "lang": "transformText"
        },
        {
          "type": "markdown",
          "lang": "transforms"
        },
        {
          "type": "style-typefaces"
        }
      ]
    },
    {
      "type": "section",
      "children": [
        {
          "type": "heading",
          "lang": "wrapHeader"
        },
        {
          "type": "text",
          "lang": "wrapText"
        },
        {
          "type": "markdown",
          "lang": "wraps"
        },
        {
          "type": "example",
          "value": "truncate"
        }
      ]
    },
    {
      "type": "section",
      "children": [
        {
          "type": "up-next",
          "value": [
            "framework/alignment",
            "framework/content"
          ]
        }
      ]
    }
  ]
}
