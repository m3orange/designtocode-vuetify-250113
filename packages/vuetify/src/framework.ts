import { createDisplay, VuetifyDisplaySymbol } from './composables/display'
import { createTheme, VuetifyThemeSymbol } from './composables/theme'
import { defaultSets, VuetifyIconSymbol } from '@/composables/icons'
import { createDefaults, VuetifyDefaultsSymbol } from '@/composables/defaults'
import { createLocaleAdapter, VuetifyLocaleAdapterSymbol } from '@/composables/locale'
import { createRtl, VuetifyRtlSymbol } from '@/composables/rtl'
import { aliases, mdi } from '@/iconsets/mdi'

// Utilities
import { mergeDeep } from '@/util'

// Types
import type { App, ComponentPublicInstance, InjectionKey } from 'vue'
import type { DisplayOptions } from '@/composables/display'
import type { ThemeOptions } from '@/composables/theme'
import type { IconOptions } from '@/composables/icons'
import type { LocaleAdapter, LocaleOptions } from '@/composables/locale'
import type { RtlOptions } from '@/composables/rtl'
import type { DefaultsOptions } from '@/composables/defaults'
import { reactive } from 'vue'

export interface VuetifyInstance {}

export interface VuetifyOptions {
  components?: Record<string, any>
  directives?: Record<string, any>
  defaults?: DefaultsOptions
  display?: DisplayOptions
  theme?: ThemeOptions
  icons?: IconOptions
  locale?: (LocaleOptions & RtlOptions) | (LocaleAdapter & RtlOptions)
}

export const VuetifySymbol: InjectionKey<VuetifyInstance> = Symbol.for('vuetify')

export const useVuetify = () => {
  const vuetify = inject(VuetifySymbol)

  if (!vuetify) {
    throw new Error('Vuetify has not been installed on this app')
  }

  return vuetify
}

export const createVuetify = (options: VuetifyOptions = {}) => {
  const install = (app: App) => {
    const {
      components = {},
      directives = {},
      icons = {},
    } = options

    for (const key in directives) {
      const directive = directives[key]

      app.directive(key, directive)
    }

    for (const key in components) {
      const component = components[key]

      app.component(key, component)
    }

    const vuetify = {}

    app.provide(VuetifySymbol, vuetify)
    app.provide(VuetifyDefaultsSymbol, createDefaults(options.defaults))
    app.provide(VuetifyDisplaySymbol, createDisplay(options.display))
    app.provide(VuetifyThemeSymbol, createTheme(options.theme))
    app.provide(VuetifyIconSymbol, mergeDeep({
      defaultSet: 'mdi',
      sets: {
        ...defaultSets,
        mdi,
      },
      aliases,
    }, icons))
    const { adapter, rootInstance } = createLocaleAdapter(app, options?.locale)
    app.provide(VuetifyLocaleAdapterSymbol, adapter)
    app.provide(VuetifyRtlSymbol, createRtl(rootInstance, options?.locale))

    function inject (this: ComponentPublicInstance, key: InjectionKey<any> | string) {
      const vm = this.$

      const provides = vm.parent?.provides ?? vm.vnode.appContext?.provides

      if (provides && (key as any) in provides) {
        return provides[(key as string)]
      }
    }

    app.mixin({
      computed: {
        $vuetify () {
          return reactive({
            defaults: inject.call(this, VuetifyDefaultsSymbol),
            display: inject.call(this, VuetifyDisplaySymbol),
            theme: inject.call(this, VuetifyThemeSymbol),
            icons: inject.call(this, VuetifyIconSymbol),
            locale: inject.call(this, VuetifyLocaleAdapterSymbol),
            rtl: inject.call(this, VuetifyRtlSymbol),
          })
        },
      },
    })
  }

  return { install }
}
