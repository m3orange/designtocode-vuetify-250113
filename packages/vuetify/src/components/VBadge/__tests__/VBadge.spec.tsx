// Components
import { VBadge } from '..'

// Utilities
import { mount } from '@vue/test-utils'
import { createVuetify } from '@/framework'

describe('VBadge', () => {
  const vuetify = createVuetify()
  const mountFunction = (options?: any) => mount(VBadge, {
    ...options,
    global: { plugins: [vuetify] },
  })

  it('should render component and match snapshot', async () => {
    const wrapper = mountFunction({
      slots: {
        badge: () => <span>content</span>,
        default: () => <span>element</span>,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component without badge', async () => {
    const wrapper = mountFunction({
      props: { modelValue: false },
      slots: {
        badge: () => <span>content</span>,
        default: () => <span>element</span>,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with color prop', () => {
    const wrapper = mountFunction({
      props: { color: 'green' },
      slots: { badge: () => <span>content</span> },
    })

    const badge = wrapper.find('.v-badge__badge')
    expect(badge.classes('bg-green')).toBeTruthy()
  })

  it('should render component without transition element', () => {
    const wrapper = mountFunction({
      props: { transition: false },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
