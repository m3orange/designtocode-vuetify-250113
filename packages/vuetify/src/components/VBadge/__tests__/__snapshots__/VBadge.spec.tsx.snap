// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VBadge should render component and match snapshot with badge slot 1`] = `
<div class="v-badge">
  <div class="v-badge__wrapper">
    <span>
      element
    </span>
    <transition-stub>
      <span class="v-badge__badge bg-primary"
            aria-atomic="true"
            aria-label="locale string here"
            aria-live="polite"
            role="status"
      >
        <span>
          content
        </span>
      </span>
    </transition-stub>
  </div>
</div>
`;

exports[`VBadge should render component and match snapshot with badge slot 2`] = `
<div class="v-badge">
  <div class="v-badge__wrapper">
    <span>
      element
    </span>
    <transition-stub>
      <span class="v-badge__badge bg-primary"
            aria-atomic="true"
            aria-label="locale string here"
            aria-live="polite"
            role="status"
            style="display: none;"
      >
        <span>
          content
        </span>
      </span>
    </transition-stub>
  </div>
</div>
`;

exports[`VBadge should render component and match snapshot with badge slot 3`] = `
<div class="v-badge v-badge--floating">
  <div class="v-badge__wrapper">
    <span>
      element
    </span>
    <transition-stub>
      <span class="v-badge__badge bg-primary"
            aria-atomic="true"
            aria-label="locale string here"
            aria-live="polite"
            role="status"
      >
        <span>
          content
        </span>
      </span>
    </transition-stub>
  </div>
</div>
`;

exports[`VBadge should render component and match snapshot with badge slot 4`] = `
<div class="v-badge v-badge--dot">
  <div class="v-badge__wrapper">
    <span>
      element
    </span>
    <transition-stub>
      <span class="v-badge__badge bg-primary"
            aria-atomic="true"
            aria-label="locale string here"
            aria-live="polite"
            role="status"
      >
      </span>
    </transition-stub>
  </div>
</div>
`;

exports[`VBadge should render component and match snapshot with badge slot 5`] = `
<div class="v-badge v-badge--dot v-badge--floating">
  <div class="v-badge__wrapper">
    <span>
      element
    </span>
    <transition-stub>
      <span class="v-badge__badge bg-primary"
            aria-atomic="true"
            aria-label="locale string here"
            aria-live="polite"
            role="status"
      >
      </span>
    </transition-stub>
  </div>
</div>
`;

exports[`VBadge should render component and match snapshot with badge slot 6`] = `
<div class="v-badge v-badge--floating">
  <div class="v-badge__wrapper">
    <span>
      element
    </span>
    <transition-stub>
      <span class="v-badge__badge bg-primary"
            aria-atomic="true"
            aria-label="locale string here"
            aria-live="polite"
            role="status"
      >
        <span>
          content
        </span>
      </span>
    </transition-stub>
  </div>
</div>
`;

exports[`VBadge should render component and match snapshot with badge slot 7`] = `
<div class="v-badge">
  <div class="v-badge__wrapper">
    <span>
      element
    </span>
    <transition-stub>
      <span class="v-badge__badge bg-primary"
            aria-atomic="true"
            aria-label="locale string here"
            aria-live="polite"
            role="status"
      >
        <span>
          content
        </span>
      </span>
    </transition-stub>
  </div>
</div>
`;

exports[`VBadge should render component and match snapshot with badge slot 8`] = `
<div class="v-badge v-badge--inline">
  <div class="v-badge__wrapper">
    <span>
      element
    </span>
    <transition-stub>
      <span class="v-badge__badge bg-primary"
            aria-atomic="true"
            aria-label="locale string here"
            aria-live="polite"
            role="status"
      >
        <span>
          content
        </span>
      </span>
    </transition-stub>
  </div>
</div>
`;

exports[`VBadge should render component and match snapshot with badge slot 9`] = `
<div class="v-badge">
  <div class="v-badge__wrapper">
    <span>
      element
    </span>
    <transition-stub>
      <span class="v-badge__badge bg-primary"
            aria-atomic="true"
            aria-label="locale string here"
            aria-live="polite"
            role="status"
      >
        <span>
          content
        </span>
      </span>
    </transition-stub>
  </div>
</div>
`;

exports[`VBadge should render component and match snapshot with badge slot 10`] = `
<div class="v-badge">
  <div class="v-badge__wrapper">
    <span>
      element
    </span>
    <transition-stub>
      <span class="v-badge__badge bg-primary"
            aria-atomic="true"
            aria-label="locale string here"
            aria-live="polite"
            role="status"
      >
        <span>
          content
        </span>
      </span>
    </transition-stub>
  </div>
</div>
`;

exports[`VBadge should render component and match snapshot with badge slot 11`] = `
<div class="v-badge">
  <div class="v-badge__wrapper">
    <span>
      element
    </span>
    <transition-stub>
      <span class="v-badge__badge bg-primary"
            aria-atomic="true"
            aria-label="locale string here"
            aria-live="polite"
            role="status"
      >
        <span>
          content
        </span>
      </span>
    </transition-stub>
  </div>
</div>
`;

exports[`VBadge should render component and match snapshot without badge slot 1`] = `
<div class="v-badge">
  <div class="v-badge__wrapper">
    <transition-stub>
      <span class="v-badge__badge bg-primary"
            aria-atomic="true"
            aria-label="locale string here"
            aria-live="polite"
            role="status"
      >
        foo
      </span>
    </transition-stub>
  </div>
</div>
`;

exports[`VBadge should render component and match snapshot without badge slot 2`] = `
<div class="v-badge">
  <div class="v-badge__wrapper">
    <transition-stub>
      <span class="v-badge__badge bg-primary"
            aria-atomic="true"
            aria-label="locale string here"
            aria-live="polite"
            role="status"
      >
        <i class="bar mdi v-icon notranslate v-icon--size-default"
           aria-hidden="true"
        >
        </i>
      </span>
    </transition-stub>
  </div>
</div>
`;

exports[`VBadge should render component and match snapshot without badge slot 3`] = `
<div class="v-badge">
  <div class="v-badge__wrapper">
    <transition-stub>
      <span class="v-badge__badge bg-primary"
            aria-atomic="true"
            aria-label="locale string here"
            aria-live="polite"
            role="status"
      >
        <i class="foo mdi v-icon notranslate v-icon--size-default"
           aria-hidden="true"
        >
        </i>
      </span>
    </transition-stub>
  </div>
</div>
`;
